File: api/app/Models/MaintenanceRecord.php
<?php
// app/Models/MaintenanceRecord.php

namespace App\Models;

use App\Models\Concerns\HasTenant;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class MaintenanceRecord extends Model
{
    use HasTenant, SoftDeletes;

    public const STATUSES = ['pending', 'in_progress', 'completed', 'cancelled'];
    public const TYPES = ['periodic', 'fault', 'revision', 'annual'];

    protected $fillable = [
        'tenant_id', 'elevator_id', 'type', 'planned_date', 'started_at',
        'completed_at', 'status', 'assigned_users', 'checklist', 'materials_used',
        'technician_note', 'customer_signature_url', 'photos', 'is_recurring',
        'recurring_period', 'parent_id', 'created_by',
    ];

    protected function casts(): array
    {
        return [
            'planned_date'   => 'datetime',
            'started_at'     => 'datetime',
            'completed_at'   => 'datetime',
            'assigned_users' => 'array',
            'checklist'      => 'array',
            'materials_used' => 'array',
            'photos'         => 'array',
            'is_recurring'   => 'boolean',
        ];
    }

    public function elevator(): BelongsTo
    {
        return $this->belongsTo(Elevator::class);
    }
}

File: api-net/Models/MaintenanceRecord.cs
namespace Liftonom.Api.Models;

public class MaintenanceRecord
{
    public long Id { get; set; }
    public long TenantId { get; set; }
    public long? ElevatorId { get; set; }
    public string? Type { get; set; }
    public string Status { get; set; } = "pending";
    public DateTime? PlannedDate { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public string? AssignedUsers { get; set; } = "[]";   // jsonb: [user_id]
    public string? Checklist { get; set; } = "[]";        // jsonb
    public string? MaterialsUsed { get; set; } = "[]";    // jsonb
    public string? TechnicianNote { get; set; }
    public bool IsCritical { get; set; }
    public string? Description { get; set; }
    public string? Notes { get; set; }
    public string? CustomerSignatureUrl { get; set; }
    public string? Photos { get; set; } = "[]";           // jsonb
    public bool IsRecurring { get; set; }
    public string? RecurringPeriod { get; set; }
    public long? ParentId { get; set; }
    public long? CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    public Elevator? Elevator { get; set; }
}
